spec/jobs/notify_email_job_spec.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+RSpec.describe NotifyEmailJob, type: :job do
+  subject(:perform!) { described_class.new.perform(notification_id: notification.id) }
+
+  let(:perform_ignore_errors!) { perform! rescue nil }
+  let(:subscription) { create(:subscription, :no_callback_url) }
+  let(:notification) { create(:notification, :email, subscription: subscription, delivered_at: delivered_at, delivery_attempted_at: nil) }
+  let(:delivered_at) { nil }
+  let(:govuk_notify_api_key) { 'GOVUK_NOTIFY_API_KEY' }
+  let(:govuk_notify_template_id) { 'GOVUK_NOTIFY_TEMPLATE_ID' }
+
+  before do
+    allow(ENV).to receive(:fetch).and_call_original
+    allow(ENV).to receive(:fetch).with('GOVUK_NOTIFY_API_KEY', nil).and_return(govuk_notify_api_key)
+    allow(ENV).to receive(:fetch).with('GOVUK_NOTIFY_TEMPLATE_ID', nil).and_return(govuk_notify_template_id)
+    ActionMailer::Base.add_delivery_method :govuk_notify, GovukNotifyRails::Delivery, api_key: govuk_notify_api_key
+  end
+
+  context 'when GOVUK_NOTIFY_API_KEY env var is not set' do
+    let(:govuk_notify_api_key) { nil }
+
+    it 'raises an NoMethodError' do
+      expect { perform! }.to raise_error(NoMethodError, /undefined method `length' for nil/)
+    end
+  end
+
+  context 'when GOVUK_NOTIFY_TEMPLATE_ID env var is not set' do
+    let(:govuk_notify_template_id) { nil }
+
+    it 'raises an ArgumentError' do
+      expect { perform! }.to raise_error(ArgumentError, /Missing template ID/)
+    end
+  end
+
+  context 'when notification and subscription are active' do
```

Comment:
this context is empty, usually `rubocop` complain about it, 🤔

Reply:
the context isn't empty? it contains multiple sub-contexts. It is intended to mark the section when notifications are enabled vs the section where they are not.